plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaHiveTypeTranslator.java

```diff
@@ -122,8 +122,8 @@ public static TypeInfo translate(Type type)
             verify(((TimestampWithTimeZoneType) type).getPrecision() == 3, "Unsupported type: %s", type);
             return HIVE_TIMESTAMP.getTypeInfo();
         }
-        if (type instanceof TimestampType) {
-            verify(((TimestampType) type).getPrecision() == 3, "Unsupported type: %s", type);
+        if (type instanceof TimestampType timestampType) {
+            verify(timestampType.getPrecision() == 3 || timestampType.getPrecision() == 6, "Unsupported type: %s", type);
```

Comment:
Which Delta type is mapped to Trino `timestamp(3)`?
It looks like none, so this used to be a dead code (copied from Hive). 
We could remove it as a prep step, and then add with `verify(precision==6...)`.

Reply:
Some tests start failing without millis precision support. Reverted the change though I haven't yet investigated the details. 
https://github.com/trinodb/trino/actions/runs/5542115619/jobs/10117009966?pr=17502